game/Board.ts

```diff
@@ -135,10 +140,55 @@ export class Board {
         return this.t[l.r][l.c];
     }
 
+    /*
+     * Getter for Tile and all of its neighbors at a location.
+     */
+    public getNeighborhood(l: Location): Tile[] {
+        let tiles: Tile[] = [];
+        for (let r = l.r - 1; r <= l.r + 1; r++) {
+            for (let c = l.c - 1; c <= l.c + 1; c++) {
+                if (this.inBounds(r, c)) {
+                    tiles.push(this.t[r][c]);
+                }
+            }
+        }
+        return tiles;
+    }
+
+    public getNearbyLocations(l: Location): Location[] {
+        let locs: Location[] = [];
+        for (let r = l.r - 1; r <= l.r + 1; r++) {
+            for (let c = l.c - 1; c <= l.c + 1; c++) {
+                if (this.inBounds(r, c)) {
+                    locs.push({ r, c });
+                }
+            }
+        }
+        return locs;
+    }
+
     /*
      * Set location to new Tile value. Enclave-only func.
      */
     public setTile(tl: Tile) {
+        const oldTile = this.t[tl.loc.r][tl.loc.c];
+        if (oldTile.owner != tl.owner) {
+            // Remove tile from old player and give to new player
+            const oldPubKey = oldTile.owner.bjjPub.serialize();
+            const index = this.playerTiles.get(oldPubKey)?.indexOf(tl.loc);
```

Comment:
can we have a diff data structure for storing player tiles

this is O(N) no? we need to be really careful about any ops that aren't O(logN)

Reply:
yeah this needs to change. another problem (which I'm only seeing rn) is you can't look for a location in an array of locations, bc typescript does equality by reference. it might be working here, but not for the reason we're expecting. Can we talk about a solution in person?

Reply:
for now i'll make a note of it